rules/aip0191/file_layout.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0191
+
+import (
+	"github.com/googleapis/api-linter/lint"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var fileLayout = &lint.FileRule{
+	Name: lint.NewRuleName(191, "file-layout"),
+	LintFile: func(f *desc.FileDescriptor) (problems []lint.Problem) {
+		// Verify that services precede messages.
+		if len(f.GetMessageTypes()) > 0 {
+			firstMessage := f.GetMessageTypes()[0]
+			for _, service := range f.GetServices() {
+				if isAfter(firstMessage, service) {
+					problems = append(problems, lint.Problem{
+						Message:    "Services should precede all messages.",
+						Descriptor: service,
+					})
+				}
```

Comment:
Do we want to break here, like the case below?

Reply:
I considered it and decided against it in this case. The number of services is smaller than the number of messages, and it really does make sense to warn on them all.

The fundamental difference is that with this warning, the _service_ is in the wrong place. With the other one, it might be that the enum is in the wrong place, and it might be messages, depending on the situation, and flagging every message as a way to indicate the need to move an enum (or vice versa) is annoying.

Reply:
Thanks!